homeassistant/components/zwave_js/climate.py

```diff
@@ -479,3 +482,72 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
         if preset_mode_value is None:
             raise ValueError(f"Received an invalid preset mode: {preset_mode}")
         await self.info.node.async_set_value(self._current_mode, preset_mode_value)
+
+
+class HeatitZTRM3Climate(ZWaveClimate):
```

Comment:
We need to discuss if this is the right approach. Keeping the device specifics in the discovery is fine, but as soon as we add device specific Entity details it's getting messy and we'll have a hard time to decline more device specific details.

Reply:
That's a fair concern and I'm happy to discuss further. FWIW I couldn't think of another way to support this device easily without introducing a lot more complexity into the base class that would still end up being device specific

Reply:
OK I tried a new approach that's more generic. Let me know what you think?